rotkehlchen/chain/ethereum/modules/arbitrum_one_bridge/decoder.py

```diff
@@ -0,0 +1,172 @@
+BRIDGE_CALL_TRIGGERED = b'-\x9d\x11^\xf3\xe4\xa6\x06\xd6\x98\x91;\x1e\xae\x83\x1a<\xdf\xe2\r\x9a\x83\xd4\x80\x07\xb0RgI\xc3\xd4f'  # noqa: E501
+
+
+class ArbitrumOneBridgeDecoder(DecoderInterface):
+    def __init__(
+            self,
+            evm_inquirer: 'EthereumInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.eth = A_ETH.resolve_to_crypto_asset()
+
+    def _decode_eth_deposit_withdraw(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes ETH deposit and withdraw events. (Bridging ETH from and to ethereum)"""
+        asset = self.eth
+        if context.tx_log.topics[0] == MESSAGE_DELIVERED:  # ETH_DEPOSIT_INITIATED
+            amount = from_wei(FVal(context.transaction.value))
+            user_address = context.transaction.from_address
+            expected_event_type = HistoryEventType.SPEND
+            new_event_type = HistoryEventType.DEPOSIT
+            expected_location_label = user_address
+            from_chain, to_chain = 'ethereum', 'arbitrum_one'
+        else:  # ETH_WITHDRAWAL_FINALIZED
+            raw_amount = hex_or_bytes_to_int(context.tx_log.data[:32])
+            user_address = hex_or_bytes_to_address(context.tx_log.topics[2])
+            amount = asset_normalized_value(raw_amount, asset)
+            expected_event_type = HistoryEventType.RECEIVE
+            new_event_type = HistoryEventType.WITHDRAWAL
+            expected_location_label = user_address
+            from_chain, to_chain = 'arbitrum_one', 'ethereum'
+
+        # Find the corresponding transfer event and update it
+        for event in context.decoded_events:
+            if (
+                event.event_type == expected_event_type and
+                event.location_label == expected_location_label and
+                event.asset == asset and
+                event.balance.amount == amount
+            ):
+                event.event_type = new_event_type
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.counterparty = CPT_ARBITRUM_ONE
+                event.notes = (
+                    f'Bridge {amount} {asset.symbol} from {from_chain} address {user_address} '
+                    f'to {to_chain} address {user_address} via arbitrum_one bridge'
+                )
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_erc20_deposit_withdraw(self, tx_log: 'EvmTxReceiptLog', decoded_events: list['EvmEvent']) -> DecodingOutput:  # noqa: E501
+        """Decodes ERC20 deposits and withdrawals. (Bridging ERC20 tokens from and to ethereum)"""
+        # Read information from event's topics & data
+        ethereum_token_address = hex_or_bytes_to_address(tx_log.data[:32])
+        asset = EvmToken(evm_address_to_identifier(
```

Comment:
I just also noticed this. Never do this.

You are trying to create a DB token from a random address we get from the internet through a decoded event. The way to do this is to use `get_or_create_evm_token`